### ISSUE TITLE
Addon open/login fails with KeyError 'lolomo'

### ISSUE THREAD

**Issue by pkscout** [CastagnaIT/plugin.video.netflix#701]
#### Your Environment
- Netflix add-on version: 1.4.1
- Operating system version/name: MacOSX
- Device model:  

Used Operating system:
* [ ] Android
* [ ] iOS
* [ ] Linux
* [x] OSX
* [ ] Raspberry-Pi
* [ ] Windows

### Describe the bug

Login fails with KeyError.

https://paste.kodi.tv/awisikaboh.kodi

#### Expected behavior

Expect to be able to login.

#### Actual behavior

I get an error dialog box.

#### Steps to reproduce the behavior

1. Install addon
2. Run addon
3. Try and login

### Debug log

The debug log can be found from this link:

#### Installation
* [ ] I'm using other Netflix Repo 
* [ ] I'm using a different source

**Comment by oss001** [comment]
Started having the same issue today.  Netflix plugin Version 1.4.1 running on Linux Elementary OS, Kodi 18.7.

**Comment by spaghetus** [comment]
Also having this issue on a Pi 4 running LibreELEC that was first installed yesterday and working earlier today.

**Comment by pkscout** [comment]
I'm not sure where the data for this check is gotten, but I was able to get the addon to load again by bypassing the isLocoSupported check in websites.py.  Specifically line 105 I changed:

`if user_data.get('isLocoSupported'):`

to 

`if True:`

That forces the added to use loco instead of lolomo (whatever that means).  At least for me that allows the addon to load as normal.  Unfortunately after that nothing plays, so whatever the issue is, it's deeper in Netflix.

**Comment by Axeinator** [comment]
Same here, OSMC, Kodi 18.7, Netflix 1.4.1

**Comment by wagnerch** [comment]
I don't know if both are required, but in:
./resources/lib/services/nfsession/nfsession_requests.py

Changed line 159:
'isLocoSupported': 'false'

to
'isLocoSupported': 'true'


And in ./resources/lib/api/website.py:
before line 104 add:
    user_data['isLocoSupported'] = 'true'   # ADD LINE
    g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
    if user_data.get('isLocoSupported'):


I am not really familiar with the details here, but I am guessing Netflix maybe rolled out this new methodology and switched from lolomo to loco which @CastagnaIT has done work on already.  Maybe if you guys have the skills you can try this kludge out until he can deliver a proper fix.

**Comment by wagnerch** [comment]
It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false.

**Comment by Axeinator** [comment]
> ./resources/lib/services/nfsession/nfsession_requests.py
> And in ./resources/lib/api/website.py:

What is the path to these on Linux (OSMC, Pi 3B+)?

**Comment by wagnerch** [comment]
Likely in your /home/osmc/.kodi folder, you can use find to see.

osmc@osmc:~$ find /home/osmc -name website.py
/home/osmc/.kodi/addons/plugin.video.netflix/resources/lib/api/website.py

I haven't tried it on my OSMC box, using Windows at the moment.

**Comment by pkscout** [comment]
> It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false. Yeah it appears this will work as well:
> 
> in ./resources/lib/api/website.py:
> before line 104 add:
> user_data['isLocoSupported'] = 'false' # ADD LINE
> g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
> if user_data.get('isLocoSupported'):

That fixed the KeyError on my Mac, but not playback.  On my OSMC device it does seem to work now (although in my brief tests I can only get 540p streams - I'll look at that next).  Now that I think about it, Netflix has never worked on my Mac.

**Comment by wagnerch** [comment]
Yeah 540p is another issue, seems to only affect ARM devices and it something on the Netflix end.

**Comment by npmstart-pray** [comment]
also happening on my RasPi4 running current xbian with addon v1.4.1
the WideVine update yesterday wouldn't have anything to do with this, would it?

**Comment by TheSurf3r** [comment]
Also happening on my Raspberry Pi 3 B with LE9.2 and on PC with Kodi 18.7:
![image](https://user-images.githubusercontent.com/12992826/84961799-de455b80-b10d-11ea-9f86-d275dbfd91d7.png)

**Comment by RJ3** [comment]
KeyError with Pi4 on LibreELEC 9.2.1.
Plugin version 1.4.1

**Comment by juandeargentina** [comment]
Same error here. Windows 10 Kodi Leia. Working fine with no changes yesterday.

**Comment by Xaque8787** [comment]
Same issue on pi 3 with both LibreELEC and osmc as OS. Got it working by reverting to [here](https://github.com/xLAva/plugin.video.netflix/releases/tag/v0.16.1) LAva version 0.16.1 on LibreELEC, guess I will run with that until this gets fixed

**Comment by npmstart-pray** [comment]
> It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false. Yeah it appears this will work as well:
> 
> in ./resources/lib/api/website.py:
> before line 104 add:
> user_data['isLocoSupported'] = 'false' # ADD LINE
> g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
> if user_data.get('isLocoSupported'):

this worked for me.  Thank you.

**Comment by Axeinator** [comment]
> I'm not sure where the data for this check is gotten, but I was able to get the addon to load again by bypassing the isLocoSupported check in websites.py. Specifically line 105 I changed:
> 
> `if user_data.get('isLocoSupported'):`
> 
> to
> 
> `if True:`
> 
> That forces the added to use loco instead of lolomo (whatever that means). At least for me that allows the addon to load as normal. Unfortunately after that nothing plays, so whatever the issue is, it's deeper in Netflix.



> It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false. Yeah it appears this will work as well:
> 
> in ./resources/lib/api/website.py:
> before line 104 add:
> user_data['isLocoSupported'] = 'false' # ADD LINE
> g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
> if user_data.get('isLocoSupported'):


Yeah, looks like neither of those are working for me, hopefully we get an update fixing this soon. I'm running OSMC, Rpi 3B+, Kodi 18.7.

**Comment by CastagnaIT** [comment]
@pkscout you have not follow the instruction to get a right log
i need a new log

**Comment by maurerr** [comment]
here's my error log
```
2020-06-18 08:41:09.270 T:1470092160   ERROR: [plugin.video.netflix (0)] Could not extract [u'models', u'truths', u'data', u'isLocoSupported']
2020-06-18 08:41:09.439 T:1470092160   ERROR: [plugin.video.netflix (0)] IPC callback raised exception: u'lolomo'
2020-06-18 08:41:09.440 T:1470092160   ERROR: [plugin.video.netflix (0)] Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/ipc.py", line 185, in make_return_call
                                                result = call(instance, func, data)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/ipc.py", line 209, in call
                                                return func(instance)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/services/nfsession/nfsession_base.py", line 37, in ensure_login
                                                return func(*args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/services/nfsession/nfsession.py", line 86, in fetch_initial_page
                                                api_data = self.website_extract_session_data(response, update_profiles=True)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/services/nfsession/nfsession.py", line 275, in website_extract_session_data
                                                return website.extract_session_data(content, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/logging.py", line 148, in timing_wrapper
                                                return func(*args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/api/website.py", line 124, in extract_session_data
                                                lolomo_root = falcor_cache['lolomo']['value'][1]
                                            KeyError: u'lolomo'
2020-06-18 08:41:09.444 T:1807721344   ERROR: [plugin.video.netflix (17)] IPC call fetch_initial_page returned KeyError: u'lolomo'
2020-06-18 08:41:09.446 T:1807721344   ERROR: [plugin.video.netflix (17)] Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/run_addon.py", line 207, in run
                                                route([part for part in g.PATH.split('/') if part])
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/run_addon.py", line 38, in lazy_login_wrapper
                                                return func(*args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/run_addon.py", line 72, in route
                                                _execute(nav_handler, pathitems[1:], g.REQUEST_PARAMS)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/run_addon.py", line 100, in _execute
                                                executor(pathitems=pathitems)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/navigation/directory.py", line 59, in root
                                                common.make_call('fetch_initial_page')
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/logging.py", line 148, in timing_wrapper
                                                return func(*args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/ipc.py", line 88, in make_call
                                                return make_http_call(callname, data)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/ipc.py", line 118, in make_http_call
                                                _raise_for_error(callname, result)
                                              File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/ipc.py", line 175, in _raise_for_error
                                                raise Exception(result['error'])
                                            Exception: KeyError
2020-06-18 08:41:11.227 T:1626149760   ERROR: GetDirectory - Error getting plugin://plugin.video.netflix/
2020-06-18 08:41:11.244 T:1937674256   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.netflix/) failed
```

**Comment by CastagnaIT** [comment]
@maurerr follow the rules or all your post will be banned

**Comment by CastagnaIT** [comment]
well, seem that today netflix roll out the website changes also to my country,
i will check better what changes is needed

**Comment by jeff2900** [comment]
Same issue for me. 
Rolback version 1.1.1 is ok under windows 10 and android 9.0.

**Comment by stevenhoving** [comment]
I can confirm that the plugin loads and functions correctly when changing line 105 to `if True:`

**Comment by enrique131** [comment]
> I'm not sure where the data for this check is gotten, but I was able to get the addon to load again by bypassing the isLocoSupported check in websites.py. Specifically line 105 I changed:
> 
> `if user_data.get('isLocoSupported'):`
> 
> to
> 
> `if True:`
> 
> That forces the added to use loco instead of lolomo (whatever that means). At least for me that allows the addon to load as normal. Unfortunately after that nothing plays, so whatever the issue is, it's deeper in Netflix.

this seems to work for me. 
Thanks

**Comment by stevenhoving** [comment]
I advice people to wait for #703 to be merged and update the plugin. #703 resolves a bit more than just changing line number 105.

**Comment by CastagnaIT** [comment]
I have temporary fixed the error on app opening, but i think that netflix will make Lolomo lists deprecated, the signals are there.

This means that in the next few days or weeks (...when netflix will change...) also the main menu will be completely non-working and also other menu lists.

More extensive modifications are required, i will be performing some tests in the next days to understand better way to get the main menu

I attach a temporary fix but **WARNING:**
- Can happen that raise unexpected errors
- If you use "Sync of watched status with netflix" can happen that stop working or raise errors

[plugin.video.netflix_1.4.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797717/plugin.video.netflix_1.4.1_20200618_t1.zip)
[plugin.video.netflix_1.4.1+matrix.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797719/plugin.video.netflix_1.4.1%2Bmatrix.1_20200618_t1.zip)

**Comment by dranidis** [comment]
Thanks for the quick update!

I installed the fix for Kodi18.x.
Now I can enter (after logging in Netflix) and see my profile and lists but when I play any movie I get the following error:
Playback error: Email or password is incorrect
and it logs me out (need  to log in again and still does not work).

Debug log: http://ix.io/2pum

**Comment by dfdario** [comment]
Great work. WAF restored! Thanks

**Comment by stevenhoving** [comment]
> Great work. WAF restored! Thanks

yup WAF levels are restored to normal

**Comment by MartChic** [comment]
> 
> 
> Great work. WAF restored! Thanks

What is WAF?

**Comment by jotakswe** [comment]
@MartChic 

Wife Acceptance Factor

**Comment by grisu48** [comment]
Thanks for the work! Had the same issue with LibreELEC on a Raspberry Pi 4. I can confirm that the hotfix is working. Thanks a lot! :+1:

**Comment by vid0** [comment]
worked for me as well, thanks! 💪 

raspberryPi3, kodi 18.6

**Comment by MichalOc** [comment]
It works like a charm for 18.x Leia - raspi retropie 3b+.
Many thanks - my little son can watch again his fav 'shaun thr sheep' ☺

**Comment by dandygig** [comment]
> ## Bug report
> #### Your Environment
> * Netflix add-on version: 1.4.1
> * Operating system version/name: LibreELEC
> * Device model: Raspi 4
> 
> Used Operating system:
> 
> * [ ]  Android
> * [ ]  iOS
> * [ ]  Linux
> * [ ]  OSX
> * [x]  Raspberry-Pi
> * [ ]  Windows
> 
> ### Describe the bug
> Login fails with KeyError.
> 
> 
> #### Expected behavior
> Expect to be able to login.
> 
> #### Actual behavior
> I get an error dialog box.
> 
> #### Steps to reproduce the behavior
> 1. Start Netflix addon
> 2. Error appears
>
> #### Possible fix
> 
> ### Additional context or screenshots (if appropriate)
> #### Installation
> * [x]  I'm using other Netflix Repo
> * [ ]  I'm using a different source
> 
> #### Other information
> #### Screenshots

**Comment by dandygig** [comment]
> > ## Bug report
> > #### Your Environment
> > 
> > * Netflix add-on version: 1.4.1
> > * Operating system version/name: LibreELEC 9.2.3
> > * Device model: Raspi 4
> > * Kodi version: 18.7.1 (Git:Leia-pi4_18.7.1.-Leia)
> > 
> > Used Operating system:
> > 
> > * [ ]   Android
> > * [ ]   iOS
> > * [ ]   Linux
> > * [ ]   OSX
> > * [x]   Raspberry-Pi
> > * [ ]   Windows
> > 
> > ### Describe the bug
> > Login fails with KeyError.
> > #### Expected behavior
> > Expect to be able to login.
> > #### Actual behavior
> > I get an error dialog box.


Update with: **Kodi 18.x: plugin.video.netflix_1.4.1_20200618_t1.zip** did help and playback is working again, no issues so far.

**Comment by CraigHB** [comment]
> Kodi 18.x: [plugin.video.netflix_1.4.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797717/plugin.video.netflix_1.4.1_20200618_t1.zip)

Hotfix version working for CoreElec 9.2.3 on Odroid N2.  Thanks for that!

**Comment by showmak** [comment]
Version 1.4.1, I am also getting KeyError message on Nvidia Shield, Kodi 18.7.

Yesterday was working though.

**Comment by adamhotep** [comment]
This new version worked for me as well. I hope Netflix better supports this sort of thing in the future and that this temporary fix can be more permanent (otherwise they may lose a customer).

Here's a reminder of [how to install Kodi add-ons from zip files](https://kodi.wiki/view/HOW-TO:Install_add-ons_from_zip_files).

The Kodi 18.x file I installed had the following stats:
```
filename: plugin.video.netflix_1.4.1_20200618_t1.zip
size:     1497878
md5:      3ae0169754c955dfe617b4393104968f
sha1:     1dabc3e78f3d4ec95dbae6d944aee5f94eaa59dc
sha256:   5a6e76818f00921b9029a60dfe807048ff1ecfcdf5e5d0a87c4d53bd3aa2c005
```

**Comment by CraigHB** [comment]
> I hope Netflix better supports this sort of thing in the future and that this temporary fix can be more permanent (otherwise they may lose a customer).

Kind of a moving target, Netflix is probably not very concerned with supporting non-mainstream devices.  I can tolerate it as long as the guys graciously supporting the add-on can keep up.  Only lost a day of use.  I think this is only the second time I've had trouble with the add-on since I've been using it.

**Comment by airaqi** [comment]
> It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false. Yeah it appears this will work as well:
> 
> in ./resources/lib/api/website.py:
> before line 104 add:
> user_data['isLocoSupported'] = 'false' # ADD LINE
> g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
> if user_data.get('isLocoSupported'):

This fix worked for me as well, thanks

**Comment by chile1900** [comment]
The fix worked on leia 18.7 on windows 10.
Thanks :)

**Comment by sergi-rollan** [comment]
Funciono en Raspberry pi3 kodi 18.7 en libreelec 9.2.3

**Comment by nasi-be-guling** [comment]
> It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false. Yeah it appears this will work as well:
> 
> in ./resources/lib/api/website.py:
> before line 104 add:
> user_data['isLocoSupported'] = 'false' # ADD LINE
> g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
> if user_data.get('isLocoSupported'):

Thanks for the quick fix. I try this and succeed.

**Comment by wagnerch** [comment]
For folks that might have overlooked it, there is a fix from the author in this comment.
https://github.com/CastagnaIT/plugin.video.netflix/issues/701#issuecomment-645910348

**Comment by showmak** [comment]
Yes the fix worked on 18.7

**Comment by Froleatric** [comment]
Thanks a lot. Fix is working in Kodi 18.7, Ubuntu 18.04.

**Comment by CastagnaIT** [comment]
I'm happy that the temporary fix works,
but this is not a forum where each one can talk freely, this is a development section

therefore post not related to the problem will be moderated
to give space to those who really have problems,
thanks to all

**Comment by CastagnaIT** [comment]
> Thanks for the quick update!
> 
> I installed the fix for Kodi18.x.
> Now I can enter (after logging in Netflix) and see my profile and lists but when I play any movie I get the following error:
> Playback error: Email or password is incorrect
> and it logs me out (need to log in again and still does not work).
> 
> Debug log: http://ix.io/2pum

This problem is not related to this issue,
You can try follow instruction on wiki:
https://github.com/CastagnaIT/plugin.video.netflix/wiki/FAQ-%28Errors%29#email-or-password-is-incorrect

or also follow the issue thread already opened with title:
_Can't login on Netflix plugin - Incorrect password_

**Comment by ThomasPause** [comment]
> # Temporary update fix
> 
> I have temporary fixed the error on app opening, but i think that netflix will make Lolomo lists deprecated, the signals are there.
> 
> This means that in the next few days or weeks (...when netflix will change...) also the main menu will be completely non-working and also other menu lists.
> 
> More extensive modifications are required, i will be performing some tests in the next days to understand better way to get the main menu
> 
> I attach a temporary fix but **WARNING:**
> 
>     * Can happen that raise unexpected errors
> 
>     * If you use "Sync of watched status with netflix" can happen that stop working or raise errors
> 
> 
> Kodi 18.x: [plugin.video.netflix_1.4.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797717/plugin.video.netflix_1.4.1_20200618_t1.zip)
> Kodi 19.x: [plugin.video.netflix_1.4.1+matrix.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797719/plugin.video.netflix_1.4.1%2Bmatrix.1_20200618_t1.zip)

works nice for the moment, thx for your work!!

**Comment by guillaume-lec** [comment]
Hey
Fix works perfectly on CoreELEC 9.2.3
Thanks a lot

**Comment by IvanBT22** [comment]
the fix worked wonderfully, excellent work @CastagnaIT

**Comment by dranidis** [comment]
> > Thanks for the quick update!
> > I installed the fix for Kodi18.x.
> > Now I can enter (after logging in Netflix) and see my profile and lists but when I play any movie I get the following error:
> > Playback error: Email or password is incorrect
> > and it logs me out (need to log in again and still does not work).
> > Debug log: http://ix.io/2pum
> 
> This problem is not related to this issue,
> You can try follow instruction on wiki:
> https://github.com/CastagnaIT/plugin.video.netflix/wiki/FAQ-%28Errors%29#email-or-password-is-incorrect
> 
> or also follow the issue thread already opened with title:
> _Can't login on Netflix plugin - Incorrect password_

Thanks for the reply. 

I managed to fix the problem. It was not related to anything mentioned in the suggested thread, though. 
I noticed that there was a trailing space at the end of my email. Although, it DID log me in correctly in my profile, when I started a movie it brought the error message. Removing the trailing space resolved the issue. Weird, though.

**Comment by CastagnaIT** [comment]
thanks for the info! i will check

**Comment by matthuo333** [comment]
Why is the resolution still 960x540 after upgrading to plugin.video.netflix-1.4.1? What parameters are set?  
My environment is: Coreelec 9.2.0

**Comment by Kinotaiyh** [comment]
> # Temporary update fix
> I have temporary fixed the error on app opening, but i think that netflix will make Lolomo lists deprecated, the signals are there.
> 
> This means that in the next few days or weeks (...when netflix will change...) also the main menu will be completely non-working and also other menu lists.
> 
> More extensive modifications are required, i will be performing some tests in the next days to understand better way to get the main menu
> 
> I attach a temporary fix but **WARNING:**
> 
> * Can happen that raise unexpected errors
> * If you use "Sync of watched status with netflix" can happen that stop working or raise errors
> 
> Kodi 18.x: [plugin.video.netflix_1.4.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797717/plugin.video.netflix_1.4.1_20200618_t1.zip)
> Kodi 19.x: [plugin.video.netflix_1.4.1+matrix.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797719/plugin.video.netflix_1.4.1%2Bmatrix.1_20200618_t1.zip)

This is really help! Thank you, it solved my problem!

**Comment by vvp2014** [comment]
This is to confirm that the hotfix  works on Debian 10.4 with Kodi 18.6. Thank you for addressing the issue promptly.

**Comment by beatschubser** [comment]
Will the plugin still receive updates after installing the temporary fix?

**Comment by matthuo333** [comment]
> > # Temporary update fix
> 
> > I have temporary fixed the error on app opening, but i think that netflix will make Lolomo lists deprecated, the signals are there.
> 
> > 
> 
> > This means that in the next few days or weeks (...when netflix will change...) also the main menu will be completely non-working and also other menu lists.
> 
> > 
> 
> > More extensive modifications are required, i will be performing some tests in the next days to understand better way to get the main menu
> 
> > 
> 
> > I attach a temporary fix but **WARNING:**
> 
> > 
> 
> > * Can happen that raise unexpected errors
> 
> > * If you use "Sync of watched status with netflix" can happen that stop working or raise errors
> 
> > 
> 
> > Kodi 18.x: [plugin.video.netflix_1.4.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797717/plugin.video.netflix_1.4.1_20200618_t1.zip)
> 
> > Kodi 19.x: [plugin.video.netflix_1.4.1+matrix.1_20200618_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4797719/plugin.video.netflix_1.4.1%2Bmatrix.1_20200618_t1.zip)
> 
> 
> 
> This is really help! Thank you, it solved my problem!


Can you achieve at least 720p resolution after updating plugin.video.netflix_1.4.1_20200618_t1.zip? Why is mine only 960*540?

**Comment by jlckodi** [comment]
> > It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false. Yeah it appears this will work as well:
> > in ./resources/lib/api/website.py:
> > before line 104 add:
> > user_data['isLocoSupported'] = 'false' # ADD LINE
> > g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
> > if user_data.get('isLocoSupported'):
> 
> That fixed the KeyError on my Mac, but not playback. On my OSMC device it does seem to work now (although in my brief tests I can only get 540p streams - I'll look at that next). Now that I think about it, Netflix has never worked on my Mac.

I have applied this workaround and it works well.
Thank you.

> > It might be the issue is just isLocoSupported was removed, so maybe the hack is to force a value to false. Yeah it appears this will work as well:
> > in ./resources/lib/api/website.py:
> > before line 104 add:
> > user_data['isLocoSupported'] = 'false' # ADD LINE
> > g.LOCAL_DB.set_value('is_loco_supported', user_data.get('isLocoSupported'), TABLE_SESSION)
> > if user_data.get('isLocoSupported'):
> 
> this worked for me. Thank you.

I have applied this workaround  and it works well. Now the menus, lists, video views, etc, running fine.

Thank you.

**Comment by CastagnaIT** [comment]
New add-on release is available

**Comment by eflukx** [comment]
@CastagnaIT Great to have a new release so fast! 💨  I know what you did this weekend.. ;) Plz. don't forget to close this issue 😉  (Just installed the patch, only then found out there was a new release)

**Comment by VaultDeveloper** [comment]
Thanks for your speed. 👍

**Comment by oss001** [comment]
Agreed! Thank you for getting to this so quickly!

________________________________

**Comment by rshpeley** [comment]
I tried to update via Kodi addons but didn't get your new release. How do I install your new release?

**Comment by Terseus** [comment]
Can confirm the version 1.5.0 is working, thank you!!

> I tried to update via Kodi addons but didn't get your new release. How do I install your new release?

Looks like the channel isn't updated yet.

You can wait for the channel to be updated with the new release, or you can [download the zip file from the releases tab](https://github.com/CastagnaIT/plugin.video.netflix/releases) and [install the addon from the zip file](https://kodi.wiki/view/HOW-TO:Install_add-ons_from_zip_files).

**Comment by saba1970** [comment]
I can also confirm that version 1.5 has solved this issue. Everything is working fine again.
@CastagnaIT thanjk you very much !!!!

**Comment by adamhotep** [comment]
> Will the plugin still receive updates after installing the temporary fix?

My system (Ubuntu 20.04, Kodi 18.7 via [team-xbmc PPA](http://ppa.launchpad.net/team-xbmc/ppa/ubuntu)) successfully auto-updated to Netflix add-on version 1.5 from the manually-installed .zip update in this thread. Everything looks good. Thanks, @CastagnaIT!

**Comment by rshpeley** [comment]
The channel is updated now. 1.5.0 is all good. Thanks for your help.
--

**Comment by jakey1995abc** [comment]
When can we expect a patch for Matrix (Kodi 19)?

**Comment by faliagas** [comment]
> Hey
> Fix works perfectly on CoreELEC 9.2.3
> Thanks a lot

How did you apply the fix?
